efficientdet/inference.py

```diff
@@ -161,11 +161,19 @@ def build_model(model_name: Text, inputs: tf.Tensor, **kwargs):
     (cls_outputs, box_outputs): the outputs for class and box predictions.
     Each is a dictionary with key as feature level and value as predictions.
   """
-  model_arch = det_model_fn.get_model_arch(model_name)
   mixed_precision = kwargs.get('mixed_precision', None)
   precision = utils.get_precision(kwargs.get('strategy', None), mixed_precision)
-  cls_outputs, box_outputs = utils.build_model_with_precision(
-      precision, model_arch, inputs, False, model_name, **kwargs)
+
+  if kwargs.get('use_keras_model', None):
```

Comment:
@fsx950223 @Ely-S @VicariNishimura 

Hi, I am trying to use keras model as default for inference, but it seems like it doesn't produce the expected results.

```
python model_inspect.py --runmode=saved_model --model_name=efficientdet-d0 \
  --hparams='image_size=1920x1280,use_keras_model=true' \
  --ckpt_path=efficientdet-d0  --saved_model_dir=/tmp/savedmodel

python model_inspect.py --runmode=saved_model_infer --model_name=efficientdet-d0 \
  --input_image=img.png --output_image_dir=/tmp/imgout/ \
  --saved_model_dir=/tmp/savedmodel/
```
It successfully loads the checkpoint and generate an output image, but the output image has very few boxes like this:

![0](https://user-images.githubusercontent.com/6027221/85504035-f995e800-b59f-11ea-8bd8-310985d70e5d.jpg)

Do you have any idea what's the problem here? Thanks!

Reply:
It seems something wrong with image_size.

Reply:
I also thought that's about image size, then I tried  --hparams='use_keras_model=true' and got zero box. So I guess there might be some other issues.

Reply:
> I also thought that's about image size, then I tried --hparams='use_keras_model=true' and got zero box. So I guess there might be some other issues.

I fix the bug by flowing changes:
```python
    def model_arch(feats, model_name=None, config=None, **kwargs):
      if not config:
        config = hparams_config.get_efficientdet_config(model_name)
      elif isinstance(config, dict):
        config = hparams_config.Config(config)  # wrap dict in Config object
      if kwargs:
        config.override(kwargs)
      model = efficientdet_arch_keras.efficientdet(feats, model_name, config=config)
      cls_outputs, box_outputs = model.outputs[1:6], model.outputs[6:]
      cls_outputs = {index+3:output for index, output in enumerate(cls_outputs)}
      box_outputs = {index+3:output for index, output in enumerate(box_outputs)}
      return cls_outputs, box_outputs
    cls_outputs, box_outputs = utils.build_model_with_precision(
        precision, model_arch, inputs, False, model_name, **kwargs)
```
```python
def efficientdet(features, model_name=None, config=None, **kwargs):
  inputs = tf.keras.layers.Input(
      [*utils.parse_image_size(config.image_size), 3], tensor=features)
```

Reply:
> > I also thought that's about image size, then I tried --hparams='use_keras_model=true' and got zero box. So I guess there might be some other issues.
> 
> I fix the bug by following changes:
> 
> ```python
>     def model_arch(feats, model_name=None, config=None, **kwargs):
>       if not config:
>         config = hparams_config.get_efficientdet_config(model_name)
>       elif isinstance(config, dict):
>         config = hparams_config.Config(config)  # wrap dict in Config object
>       if kwargs:
>         config.override(kwargs)
>       model = efficientdet_arch_keras.efficientdet(feats, model_name, config=config)
>       cls_outputs, box_outputs = model.outputs[1:6], model.outputs[6:]
>       cls_outputs = {index+3:output for index, output in enumerate(cls_outputs)}
>       box_outputs = {index+3:output for index, output in enumerate(box_outputs)}
>       return cls_outputs, box_outputs
>     cls_outputs, box_outputs = utils.build_model_with_precision(
>         precision, model_arch, inputs, False, model_name, **kwargs)
> ```
> 
> ```python
> def efficientdet(features, model_name=None, config=None, **kwargs):
>   inputs = tf.keras.layers.Input(
>       [*utils.parse_image_size(config.image_size), 3], tensor=features)
> ```

The strange behavior is caused by placeholder.

Reply:
@fsx950223 I checked in a commit with your fix, but somehow still couldn't get the expected output. I might have missed something.
Could you take a look or check in your change to this branch (you should have write access now)? Thanks!